### DISCUSSION TITLE
Native localization

### DISCUSSION THREAD

**Discussion by J3ster1337** [fgilde/MudBlazor.Extensions#37] [Ideas]
Do you interested in providing embedded localization for components that have text in them (file preview, file download, etc)?
If yes, can I contribute to creating Russian localization? 
If yes, how can I do it?

**Comment by fgilde** [comment]
In general every component supports localization already. In your Project you can add any Localization as you want like *resx different razor files or what you like. I use simple Yaml localization in mine. Automatically a IStringLocalizer<T> is injected and if available it tries to localize all texts. 
Not all are configurable right now but if you use for example in the UploadEdit the Property `TextUploadFile` and you set it to a key that is in your localization it automatically will translated.

So with this behaviour its total flexible and the whole responsibility for localization is in the project you build. In my opinion this is a clever idea :-) .. 
And it works perfect in my app for example. 

In my application I use my dotnet tool [ResourceTranslator.CLI](https://www.nuget.org/packages/ResourceTranslator.CLI#readme-body-tab) and create all all translations automatically before build. 

Not sure if it make sense to have localizations nativ in the component library then.

**Comment by J3ster1337** [comment]
Okay, thanks!

**Comment by CristianSala76** [comment]
hello, I have localization via .resx resource files, I cannot understand how to translate the MudExFileDisplay literals.
I see you use a TryLocalize function but it is not clear to me. 
Can you help me please?
thank you

**Comment by fgilde** [comment]
In general because I do everything alone in my freetime its hard to have everything documented. 
I will create a documentation and maybe include localization in the sample project as well.

You need to add localization in your project I do this with combining multiple localizers on my own but however in general its just registering a service that implements `IStringLocalizer<>`

```
      builder.Services .AddTransient(typeof(IStringLocalizer<>), typeof(YourLocalizer<>))
```

Then you can just use the current value for a string as Key and translate it as you want, or implement your own Localizer as you want

**Reply by CristianSala76** [reply]
I have managed to get it work by passing my localizer to the MudExFileDisplay ( Localizer="@localizer") aand then by putting the string to be localized in the resx file as key

**Reply by CristianSala76** [reply]
one problem I have is that the Localizer parameter is not inherited in the MudExFileDisplay  used inside a MudExFileDisplayZip:
also other parameters like AllowCopyUrl are not inherited.

**Reply by fgilde** [reply]
Oh if this is true I will fix it asap. Can you crate a bug issue for that please
